### PR TITLE
Support for custom labels

### PR DESCRIPTION
**Pull request by nlamirault**
**Is this a bug fix or adding new feature?**

**What is this PR about? / Why do we need it?**

Add custom labels to metadata for all resources

**What testing is done?**

### CODE REVIEW

File: charts/aws-ebs-csi-driver/values.yaml [hunk 1]
```diff
@@ -8,6 +8,10 @@ image:
   tag: ""
   pullPolicy: IfNotPresent
 
+# -- Custom labels to add into metadata
+customLabels: #{}
```
Comment (krmichel) [inline charts/aws-ebs-csi-driver/values.yaml:12]: I think custom labels should be empty for the default values for the chart

File: charts/aws-ebs-csi-driver/templates/_helpers.tpl [hunk 2]
```diff
@@ -41,8 +41,12 @@ helm.sh/chart: {{ include "aws-ebs-csi-driver.chart" . }}
 {{- if .Chart.AppVersion }}
 app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
 {{- end }}
+app.kubernetes.io/component: csi-driver
```
Comment (krmichel) [inline charts/aws-ebs-csi-driver/templates/_helpers.tpl:44]: I am not familiar enough with csi, in general, to know if `csi-driver` is a sufficient component name or if it should be `ebs-csi-driver`.  @wongma7 @ayberk thoughts?

**Review by krmichel — CHANGES_REQUESTED** [review]

File: charts/aws-ebs-csi-driver/values.yaml (hunk 1 above)
Comment (nlamirault) [reply charts/aws-ebs-csi-driver/values.yaml]: done

File: charts/aws-ebs-csi-driver/templates/_helpers.tpl (hunk 2 above)
Comment (wongma7) [reply charts/aws-ebs-csi-driver/templates/_helpers.tpl:44]: I think it's OK, the example from https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/ is `database` which is very generic. https://github.com/kubernetes-csi/csi-driver-host-path/blob/master/deploy/kubernetes-1.21/hostpath/csi-hostpath-driverinfo.yaml#L9 also is setting it so lgtm

**Pull request merged**